lib/rspec/rails/view_path_builder.rb

```diff
@@ -0,0 +1,53 @@
+module RSpec
+  module Rails
+    # Builds paths for view specs using a particular route set.
+    class ViewPathBuilder
+      def initialize(route_set)
+        @routes = route_set.routes
+        self.class.send(:include, route_set.url_helpers)
+      end
+
+      # Given a hash of parameters, build a view path, if possible.
+      # Returns nil if no path can be built from the given params.
+      #
+      # @example
+      #     # path can be built because all required params are present in the hash
+      #     view_path_builder = ViewPathBuilder.new(::Rails.application.routes)
+      #     view_path_builder.path_for({ :controller => 'posts', :action => 'show', :id => '54' })
+      #     # => "/post/54"
+      #
+      # @example
+      #     # path cannot be built because the params are missing a required element (:id)
+      #     view_path_builder.path_for({ :controller => 'posts', :action => 'delete' })
+      #     # => nil
+      def path_for(path_params)
+        return nil unless route_exists?(path_params)
+        url_for(path_params.merge(:only_path => true))
+      end
```

Comment:
Could we return a helpful message as the path? Something like: "No path for { hash details } could be found."

This way if someone relies on this being set, and it fails, we give them some details instead of just a `nil`.